core/src/test/scala/unit/kafka/coordinator/group/CoordinatorLoaderImplTest.scala

```diff
@@ -0,0 +1,315 @@
+    val tp = new TopicPartition("foo", 0)
+    val replicaManager = mock(classOf[ReplicaManager])
+    val serde = mock(classOf[StringKeyValueDeserializer])
+    val log = mock(classOf[UnifiedLog])
+    val coordinator = mock(classOf[CoordinatorPlayback[(String, String)]])
+
+    TestUtils.resource(new CoordinatorLoaderImpl[(String, String)](
+      replicaManager = replicaManager,
+      deserializer = serde,
+      loadBufferSize = 1000
+    )) { loader =>
+      when(replicaManager.getLog(tp)).thenReturn(Some(log))
+      when(log.logStartOffset).thenReturn(0L)
+      when(replicaManager.getLogEndOffset(tp)).thenReturn(Some(2L))
+
+      val readResult = logReadResult(startOffset = 0, records = Seq(
+        new SimpleRecord("k1".getBytes, "v1".getBytes),
+        new SimpleRecord("k2".getBytes, "v2".getBytes)
+      ))
+
+      when(log.read(
+        startOffset = 0L,
+        maxLength = 1000,
+        isolation = FetchIsolation.LOG_END,
+        minOneMessage = true
+      )).thenReturn(readResult)
+
+      when(serde.deserialize(ArgumentMatchers.any(), ArgumentMatchers.any()))
+        .thenThrow(new UnknownRecordTypeException(1))
+        .thenReturn(("k2", "v2"))
+
+      loader.load(tp, coordinator).get(10, TimeUnit.SECONDS)
+
+      verify(coordinator).replay(("k2", "v2"))
+    }
+  }
+
+  @Test
+  def testDeserializationErrorFailsTheLoading(): Unit = {
+    val tp = new TopicPartition("foo", 0)
+    val replicaManager = mock(classOf[ReplicaManager])
+    val serde = mock(classOf[StringKeyValueDeserializer])
+    val log = mock(classOf[UnifiedLog])
+    val coordinator = mock(classOf[CoordinatorPlayback[(String, String)]])
+
+    TestUtils.resource(new CoordinatorLoaderImpl[(String, String)](
+      replicaManager = replicaManager,
+      deserializer = serde,
+      loadBufferSize = 1000
+    )) { loader =>
+      when(replicaManager.getLog(tp)).thenReturn(Some(log))
+      when(log.logStartOffset).thenReturn(0L)
+      when(replicaManager.getLogEndOffset(tp)).thenReturn(Some(2L))
+
+      val readResult = logReadResult(startOffset = 0, records = Seq(
+        new SimpleRecord("k1".getBytes, "v1".getBytes),
+        new SimpleRecord("k2".getBytes, "v2".getBytes)
+      ))
+
+      when(log.read(
+        startOffset = 0L,
+        maxLength = 1000,
+        isolation = FetchIsolation.LOG_END,
+        minOneMessage = true
+      )).thenReturn(readResult)
+
+      when(serde.deserialize(ArgumentMatchers.any(), ArgumentMatchers.any()))
+        .thenThrow(new RuntimeException("Error!"))
+
+      val ex = assertFutureThrows(loader.load(tp, coordinator), classOf[RuntimeException])
+      assertEquals("Error!", ex.getMessage)
+    }
+  }
+
+  @Test
+  def testLoadGroupAndOffsetsWithCorruptedLog(): Unit = {
+    // Simulate a case where startOffset < endOffset but log is empty. This could theoretically happen
+    // when all the records are expired and the active segment is truncated or when the partition
```

Comment:
I think it makes sense to handle this silently rather than throw an error, but maybe we want to have a comment about this in the code?